pkg/istio/authorizationpolicy/controller_test.go

```diff
@@ -378,55 +402,47 @@ func TestSyncAuthzPolicy(t *testing.T) {
+				annotations := (tt.item.Resource).(*authz.AuthorizationPolicy).Annotations
+				if annotations != nil {
+					authzPolicyConfig.Annotations = annotations
 				}
+				_, err := c.configStoreCache.Create(authzPolicyConfig)
+				assert.Nil(t, err, "configstore create resource should not return error")
 				time.Sleep(100 * time.Millisecond)
-				(tt.item.Resource).(*authz.AuthorizationPolicy).ObjectMeta.ResourceVersion = resourceVersion
 				key, err := cache.MetaNamespaceKeyFunc(tt.item.Resource)
-				assert.Equal(t, nil, err, "function convert item interface to key should not return error")
+				assert.Nil(t, err, "function convert item interface to key should not return error")
 				err = c.sync(key)
-				if err != nil {
-					log.Panicln("controller has sync err: ", err)
-				}
+				assert.Nil(t, err, "sync function should not return error")
 				// Updated Spec should be equal
 				genAuthzPolicy := c.configStoreCache.Get(collections.IstioSecurityV1Beta1Authorizationpolicies.Resource().GroupVersionKind(), tt.expectedAuthzPolicy.Name, tt.expectedAuthzPolicy.Namespace)
-				assert.Equal(t, tt.expectedAuthzPolicy.Spec, genAuthzPolicy.Spec, "created authorization policy spec should be equal")
-				// except creation timestamp and revision, configMeta field should be same
-				assert.Equal(t, tt.expectedAuthzPolicy.ConfigMeta.Namespace, genAuthzPolicy.ConfigMeta.Namespace, "namespace should be equal")
-				assert.Equal(t, tt.expectedAuthzPolicy.ConfigMeta.Type, genAuthzPolicy.ConfigMeta.Type, "resource type should be equal")
-				assert.Equal(t, tt.expectedAuthzPolicy.ConfigMeta.Name, genAuthzPolicy.ConfigMeta.Name, "resource name should be equal")
-				assert.Equal(t, tt.expectedAuthzPolicy.ConfigMeta.Group, genAuthzPolicy.ConfigMeta.Group, "apiGroup should be equal")
-				assert.Equal(t, tt.expectedAuthzPolicy.ConfigMeta.Version, genAuthzPolicy.ConfigMeta.Version, "apiVersion should be equal")
+				// set creation timestamp and resource version for expected authz policy resource, which are generated on the fly
+				tt.expectedAuthzPolicy.ConfigMeta.CreationTimestamp = genAuthzPolicy.ConfigMeta.CreationTimestamp
+				tt.expectedAuthzPolicy.ConfigMeta.ResourceVersion = genAuthzPolicy.ConfigMeta.ResourceVersion
+				assert.Equal(t, tt.expectedAuthzPolicy, *genAuthzPolicy, "created authorization policy spec should be equal")
 			case model.EventDelete:
-				c := newFakeController(&adv1.AthenzDomain{}, &v1.Service{}, tt.fake, make(chan struct{}))
-				c.adIndexInformer.GetStore().Add(tt.inputAthenzDomain.DeepCopy())
-				c.serviceIndexInformer.GetStore().Add(tt.inputService.DeepCopy())
+				c := newFakeController(&adv1.AthenzDomain{}, &v1.Service{}, true, make(chan struct{}))
+				err := c.adIndexInformer.GetStore().Add(tt.inputAthenzDomain.DeepCopy())
+				assert.Nil(t, err, "add athenz domain crd to cache should not return error")
+				err = c.serviceIndexInformer.GetStore().Add(tt.inputService.DeepCopy())
```

Comment:
Should we directly pass the athenz domain and service objects into newFakeController?